game-core/src/main/java/games/strategy/engine/data/GameData.java

```diff
@@ -108,7 +108,7 @@
   private final List<Tuple<IAttachment, ArrayList<Tuple<String, String>>>> attachmentOrderAndValues =
       new ArrayList<>();
   // TODO: change to Map/HashMap upon next incompatible release
-  private final Hashtable<String, TerritoryEffect> territoryEffectList = new Hashtable<>();
+  private final Map<String, TerritoryEffect> territoryEffectList = new HashMap<>();
```

Comment:
I was about to say that this would be an incompatible change, however HashTable implements the map interface and therefore it should be backwards compatible. 
However you should probably remove the comment stating this change then

Reply:
@ssoloff I recall you added this comment here, your opinion about my theory?

Reply:
It will work when `master` loads an old save game, as you observed, because you can assign a `Hashtable` to a `Map`.  However, if an older client tries to load a save game from `master`, I think it will break because it will try to deserialize a `HashMap` into a `Hashtable` field.

Reply:
@ssoloff I'd vote to merge this then, our engine is really not very good in forward compatibility anyways

Reply:
@RoiEXLab I'm confused.  We just fixed a regression that would not allow a pre-release client to play a network game with a 3635 client.  This is the same issue.  If you host a game from this branch, it will send a `GameData` with a `HashMap` to the other nodes.  If one of the other nodes is an older client, it won't be able to deserialize the `GameData`, and the network game will not start.

Reply:
@ssoloff Right, I forgot about the network case 🙈 
My though was: If someone already upgraded, why should he downgrade again?